presto-cache/src/main/java/com/facebook/presto/cache/CachingFileSystem.java

```diff
@@ -498,4 +501,25 @@ public boolean isCacheValidationEnabled()
     {
         return cacheValidationEnabled;
     }
+
+    @Override
+    public RemoteIterator<LocatedFileStatus> listDirectory(Path path)
+            throws IOException
+    {
+        return dataTier.listDirectory(path);
+    }
+
+    @Override
+    public RemoteIterator<HiveFileInfo> listFiles(Path path)
+            throws IOException
+    {
+        return dataTier.listFiles(path);
+    }
+
+    @Override
+    public FSDataInputStream openFile(Path path, HiveFileContext hiveFileContext)
+            throws Exception
+    {
+        return new CachingInputStream(dataTier.openFile(path, hiveFileContext), cacheManager, path, cacheValidationEnabled);
```

Comment:
You need to check `hiveFileContext.isCacheable()` to decide if to cache right?

Reply:
good catch.